force-app/main/default/lwc/serviceDeliveryRow/serviceDeliveryRow.js

```diff
@@ -428,185 +347,283 @@ export default class ServiceDeliveryRow extends LightningElement {
 
             if (
                 !this.hasContactField &&
                 this.hasProgramEngagementField &&
                 this.programEngagementId
             ) {
-                this.isServiceFiltered = true;
-                this.handleGetServicesForProgramEngagement(this.programEngagementId);
+                this.getRelatedRecordsFromProgramEngagement();
             }
 
-            if (this.hasContactField && !this.hasProgramEngagementField && contactId) {
-                this.handleEnableFieldOnInputChange(this.fields.contact.fieldApiName);
-            }
+            this.setDisabledAttribute();
         }
     }
 
-    handleSaveStart() {
+    setSaving() {
         this.saveMessage = "...";
         this.isSaving = true;
         this.isSaved = false;
-        this.isError = false;
     }
 
-    handleSaveEnd() {
+    setSaved() {
         this.isSaving = false;
         this.isSaved = true;
         this.isDirty = false;
-        this.enableDisableFieldsOnSaveAndInputChange();
     }
 
-    onSave(event) {
-        if (event.detail) {
-            this.programEngagementId = event.detail;
-            if (this.selectedContact && this.programEngagementId) {
-                this.handleGetServicesEngagements(this.selectedContact);
-            }
+    setUnitOfMeasurement(fields) {
+        this.unitOfMeasureValue =
+            fields[this.fields.unitOfMeasurement] &&
+            fields[this.fields.unitOfMeasurement].value
+                ? fields[this.fields.unitOfMeasurement].value
+                : this.labels.quantity;
+    }
+
+    setCurrentFieldSet() {
+        this.fieldSet = this.serviceDeliveryFieldSets.getCurrentFieldSet().map(field => ({
+            ...field,
+        }));
+
+        this.hasContactField = this.serviceDeliveryFieldSets.hasContactField(
+            this.fieldSet
+        );
+        this.hasProgramEngagementField = this.serviceDeliveryFieldSets.hasProgramEngagementField(
+            this.fieldSet
+        );
+    }
+
+    setCurrentFieldSetName(fields) {
+        let serviceDeliveryFieldSet =
+            fields[this.fields.fieldSet] && fields[this.fields.fieldSet].value
+                ? fields[this.fields.fieldSet].value
+                : DEFAULT_FIELD_SET;
+
+        if (
+            this.serviceDeliveryFieldSets.currentFieldSetName === serviceDeliveryFieldSet
+        ) {
+            return false;
         }
+
+        this.serviceDeliveryFieldSets.currentFieldSetName = serviceDeliveryFieldSet;
+        return true;
     }
 
-    onCancel() {
-        this.template.querySelectorAll("lightning-combobox").forEach(element => {
-            element.value = null;
+    setProgramEngagementOptions() {
+        let programEngagementField = this.fieldSet.find(
+            member => member.apiName === this.fields.programEngagement
+        );
+
+        let engagements = [...this._programEngagements];
+
+        engagements.push({
+            label: "\u254B   " + newProgramEngagement,
```

Comment:
I think the code was like this before, but following our patterns shouldn't this be labels.newProgramEngagement?